app/src/main/java/com/example/tanya/godeltechchallengeandroid/di/component/AppComponent.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.tanya.godeltechchallengeandroid.di.component
+
+import com.example.tanya.godeltechchallengeandroid.App
+import com.example.tanya.godeltechchallengeandroid.di.module.ActivityBuilder
+import com.example.tanya.godeltechchallengeandroid.di.module.AppModule
+import com.example.tanya.godeltechchallengeandroid.di.module.StartModule
+import dagger.BindsInstance
+import dagger.Component
+import dagger.android.AndroidInjectionModule
+import javax.inject.Singleton
+
+@Singleton
+@Component(modules= [AppModule::class, AndroidInjectionModule::class, ActivityBuilder::class, StartModule::class])
+interface AppComponent {
```

Comment:
Here is the same... you have to implement the AndroidInjector class and the AndroidInjector.Builder to reduce code (once again - don't invent a bicycle :) )

Reply:
If you use a scope annotation (the Singleton one in our case) in modules you have to add the same scope annotation to one from components to say Dagger where it can keep a scoped cache.

Reply:
> If you use a scope annotation (the Singleton one in our case) in modules you have to add the same scope annotation to one from components to say Dagger where it can keep a scoped cache.

I don't fully undersatnd what am I supposed to change here :)

Reply:
@Tanyayasiuk Sorry I didn't see you put the @Singleton annotation on the component.